src/services/filesync/filesync.ts

```diff
@@ -736,7 +823,15 @@ export class FileSync {
    * Updates {@linkcode _syncJson} and saves it to `.gadget/sync.json`.
    */
   private async _save(filesVersion: string | bigint): Promise<void> {
-    this._syncJson = { ...this._syncJson, mtime: Date.now() + 1, filesVersion: String(filesVersion) };
+    assert(this._syncJson.environments[this.environment], "environment doesn't exist in _syncJson");
+
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    this._syncJson.environments[this.environment]!.filesVersion = String(filesVersion);
```

Comment:
Interesting, I would have assumed TypeScript would realize that `this._syncJson.environments[this.environment]` wasn't nullable since you `assert`'d it isn't.